community/detectors/localai_cve_2024_2029/src/test/java/com/google/tsunami/plugins/detectors/rce/cve20242029/LocalAiCve20242029RceDetectorTest.java

```diff
@@ -0,0 +1,200 @@
+    mockTargetService.shutdown();
+    mockCallbackServer.shutdown();
+  }
+
+  @Test
+  public void detect_whenVulnerable_returnsVulnerability_Cve202229165_Oob() throws IOException {
+    startMockWebServerForTestingWithOob(true);
+    createInjector();
+    mockCallbackServer.enqueue(PayloadTestHelper.generateMockSuccessfulCallbackResponse());
+
+    DetectionReportList detectionReports =
+        detector.detect(targetInfo, ImmutableList.of(targetNetworkService));
+
+    assertThat(detectionReports.getDetectionReportsList())
+        .containsExactly(
+            DetectionReport.newBuilder()
+                .setTargetInfo(targetInfo)
+                .setNetworkService(targetNetworkService)
+                .setDetectionTimestamp(
+                    Timestamps.fromMillis(Instant.now(fakeUtcClock).toEpochMilli()))
+                .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+                .setVulnerability(
+                    Vulnerability.newBuilder()
+                        .setMainId(
+                            VulnerabilityId.newBuilder()
+                                .setPublisher("TSUNAMI_COMMUNITY")
+                                .setValue("CVE_2024_2029"))
+                        .setSeverity(Severity.CRITICAL)
+                        .setTitle("CVE-2024-2029 LocalAI Remote Code Execution")
+                        .setRecommendation(
+                            "LocalAI user should upgrade the LocalAI to the versions v2.10.0 and"
+                                + " above.")
+                        .setDescription(
+                            "Publicly exposed LocalAI instances before v2.7.0 are vulnerable to"
+                                + " Remote Code Execution Vulnerability. Attackers can inject"
+                                + " arbitrary OS commands within the audio filename filed during"
+                                + " uploading the audio file with a POST HTTP request and send it"
+                                + " to the v1/audio/transcriptions endpoint."))
+                .build());
+    Truth.assertThat(mockTargetService.getRequestCount()).isEqualTo(3);
+    Truth.assertThat(mockCallbackServer.getRequestCount()).isEqualTo(1);
+  }
+
+  @Test
+  public void detect_ifNotVulnerable_doesNotReportVuln_Exposed_Ui() throws IOException {
+    startMockWebServerForTestingWithOob(false);
+    createInjector();
+    mockCallbackServer.enqueue(PayloadTestHelper.generateMockUnsuccessfulCallbackResponse());
+    mockCallbackServer.enqueue(PayloadTestHelper.generateMockUnsuccessfulCallbackResponse());
+    DetectionReportList detectionReports =
+        detector.detect(targetInfo, ImmutableList.of(targetNetworkService));
+    assertThat(detectionReports.getDetectionReportsList()).isEmpty();
+    Truth.assertThat(mockTargetService.getRequestCount()).isEqualTo(3);
+  }
+
+  private void startMockWebServerForTestingWithOob(boolean mustHaveForgedCookie)
```

Comment:
@devampkid mustHaveForgedCookie is not used in this method, do you remember what was it intended for?